Classes/OptimizedRedisCacheBackend.php

```diff
@@ -222,11 +219,10 @@ private function buildKey(string $identifier): string
     private function calculateExpires($tag, $lifetime)
     {
         $ttl = $this->redis->ttl($tag);
-        if ($ttl === -1 || $lifetime === self::UNLIMITED_LIFETIME) {
-            return 0;
-        } else {
-            return max($ttl, $lifetime);
+        if ($ttl < 0 || $lifetime === self::UNLIMITED_LIFETIME) {
```

Comment:
With Redis version 2.8+ `ttl` can return -2 as well (see https://redis.io/commands/ttl).
The result should be the same, but this way it is more explicit